Problema3.txt

```diff
@@ -0,0 +1,29 @@
+public bool ThereAreDuplicates(int[] a)
+        {
+            int gaussSum = 0, actualSum = 0;
+            long multiply = 1, factorial = 1;
+            int size = a.Length;
+            for (int i = 0; i < size; i++)
+            {
+                actualSum += a[i];
+                multiply *= a[i];
+                factorial *= (i + 1);
+            }
+            gaussSum = size * (size + 1) / 2;
+            if (gaussSum != actualSum)
+
+                return true;
+            else
+                if (multiply != factorial)
+                    return true;
+            return false;
+        }
+        
+          public void Arrays_WithDuplicates_ShouldFail()
+        {
+            int[] a = new int[] {2,2,6,1,3,3,5};
+            Methods method = new Methods();
```

Comment:
Pentru instructiunile de mai sus(fiind version1) returneaza ca a trecut testul pentru ca depaseste limita long-ului si produsul da 0 in ambele cazuri. La versiunea updata la care lucrez acuma, folosesc BigInteger in loc de long si se rezolva problema.